graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_label_transformation.py

```diff
@@ -63,13 +64,15 @@ def apply(self, input_df: DataFrame) -> DataFrame:
 
         # Labels that were missing and were assigned the value numLabels by the StringIndexer
         # are converted to None
-        long_class_label = indexed_df.select(F.col(self.label_column).cast("long")).select(
+        long_class_label = indexed_df.select(
             F.when(
                 F.col(self.label_column) == len(str_indexer_model.labelsArray[0]),  # type: ignore
                 F.lit(None),
             )
             .otherwise(F.col(self.label_column))
-            .alias(self.label_column)
+            .cast("long")
+            .alias(self.label_column),
+            *original_cols,
```

Comment:
So we add all the original columns here for the label value? Will it overkill?

Reply:
Hmm, well here we want to ensure that the order_col is preserved in the output, but `DistSingleLabelTransformation` is not aware of `order_col`. We could modify the constructor to also provide that, and then return just the label and order column.

The argument against would be that these two cols are selected downstream anyway, and the other columns are not materialized, so there shouldn't be any real performance penalty. 

I'm good with either option.